src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -238,6 +238,15 @@ public int Run()
                     }
                 }
 
+                if (nativeLib || SplitExeInitialization)
```

Comment:
Like Jan said in the other comment - this should not be conditioned.

Maybe we could also remove the the spot that adds ExportedMethodsRootProvider for corelib and move that to the .targets file? Treating CoreLib specially for this shouldn't be needed once we have the argument.

Reply:
I agree.
Added `System.Private.CoreLib` to the `UnmanagedEntryPointsAssembly` in the `.targets` file and removed the logic for implicitly exporting its unmanaged entry points.